### ISSUE TITLE
3-networks-hub-and-spoke Deploying with Cloud Build step 9 fails

### ISSUE THREAD

**Issue by moszutij** [terraform-google-modules/terraform-example-foundation#1045]
### TL;DR

Since updating terraform-google-modules/network/google to [v7](https://github.com/terraform-google-modules/terraform-example-foundation/commit/2f54ad621404fb2a38df0006f4692fdb0a22ee12),  deploying 3-networks-hub-and-spoke with Cloud Build Step 9 to generate the Terraform plan fails. 

### Expected behavior

The plan is generated with no errors

### Observed behavior

The following error is thrown:
*************** TERRAFORM PLAN *******************
At environment: envs/shared

│ Error: Invalid value for input variable
│ 
│   on ../../modules/base_shared_vpc/main.tf line 37, in module "main":
│   37:   subnets          = var.subnets
│ 
│ The given value is not suitable for module.base_shared_vpc.module.main.var.subnets declared at
│ .terraform/modules/base_shared_vpc.main/variables.tf:39,1-19: incorrect list element type: object required.

│ Error: Invalid value for input variable
│ 
│   on ../../modules/restricted_shared_vpc/main.tf line 37, in module "main":
│   37:   subnets          = var.subnets
│ 
│ The given value is not suitable for module.restricted_shared_vpc.module.main.var.subnets declared at
│ .terraform/modules/restricted_shared_vpc.main/variables.tf:39,1-19: incorrect list element type: object required.

### Terraform Configuration

```hcl
The out of box configurations are used
```

### Terraform Version

```sh
$ terraform --version
Terraform v1.3.0
on linux_amd64
```

### Additional information

Version 7.1.0 of the Google network subnets module defines a subnet as a [list(map(string))](https://registry.terraform.io/modules/terraform-google-modules/network/google/7.1.0/submodules/subnets#inputs) whereas version 7.2.0 defines the subnet as a [list(object({..})](https://registry.terraform.io/modules/terraform-google-modules/network/google/7.2.0/submodules/subnets)
